.github/actions/bazel-test-all/action.yaml

```diff
@@ -83,28 +83,51 @@ runs:
                 RUN_ON_DIFF_ONLY="false"
             fi
 
-            # Some environment variables needed by diff.sh
-            export BAZEL_TARGETS='${{ inputs.BAZEL_TARGETS }}'
-            export BAZEL_COMMAND='${{ inputs.BAZEL_COMMAND }}'
-
-            if [[ "${RUN_ON_DIFF_ONLY:-}" == "true" ]]; then
-                # get bazel targets that changed within the MR
-                BAZEL_TARGETS=$("${CI_PROJECT_DIR:-}"/ci/bazel-scripts/diff.sh)
-            fi
-
-            # if bazel targets is empty we don't need to run any tests
-            if [ -z "${BAZEL_TARGETS:-}" ]; then
-                echo "No bazel targets to build"
-                exit 0
-            fi
 
             bazel_args=( )
 
             if [[ $release_build == "true" ]]; then
               bazel_args+=( --config=release )
             fi
 
+            BAZEL_TARGETS='${{ inputs.BAZEL_TARGETS }}'
+
+            bazel_targets=( )
+            for tgt in $BAZEL_TARGETS; do
+              bazel_targets+=( "$tgt" )
+            done
+
+
+            if [[ "${RUN_ON_DIFF_ONLY:-}" == "true" ]]; then
+                target_pattern_file=$(mktemp) # TODO: remove (or upload?)
+                merge_base_sha='${{ github.event.pull_request.base.sha }}'
+                branch_head_sha='${{ github.event.pull_request.head.sha }}'
+
+                # we can't use HEAD as a fallback for branch_head_sha because that is
+                # the merge commit which contains the changes of the current HEAD of master
+                # XXX: the logic was ported from a legacy script though ranges like `HEAD..`
```

Comment:
What about we just make it an error now if `merge_base_sha` or `branch_head_sha` are undefined? Which means we need to ensure `RUN_ON_DIFF_ONLY` is only `true` on PRs which is currently [the case](https://sourcegraph.com/github.com/dfinity/ic/-/blob/.github/workflows/ci-main.yml?L45).